### ISSUE TITLE
Running Time for Flow Generators

### ISSUE THREAD

**Issue by aneeshbal** [jbohnslav/deepethogram#6]
Hi! This is great work, and I would like to use it for our data! I had no issues with opening the GUI and initiating a project, but when it came to training the flow generators, it took about 8 minutes for 1 iteration, and the total projected time was about 299 hours. This is obviously untenable, but I noticed in the paper it is mentioned that it should take around 24 hours? 

Do you have any idea what could be going on here? First I assumed that PyTorch was not recognizing the GPU, which is a RTX 2080, but it definitely finds it and loads variables on GPU. Also, I am running this on a standard SSD drive (C: drive). 

Let me know if there is anything I should be doing differently!

**Comment by jbohnslav** [comment]
Thanks for trying the software! Great question. I need to write a performance section at some point. 

1. Clarification: the total projected time is for 1000 epochs by default. However, training will quit automatically long before then. With default settings, it will detect when the validation loss saturates, and then decrement the learning rate. These two steps will be performed 4 times, and then training will stop. This should take less than 100 epochs (sometimes far less), resulting in something like the 24 hours. However, it's hard to predict this process, so the progress bar just assumes 1000 epochs. 
2. When you look at CPU and GPU usage on some kind of resource monitor (e.g. MSI Afterburner on Windows, or htop / nvidia-smi on Linux), is one of them maxed out? If you're using .mp4 or .avi video files, the CPU can be the bottleneck. I'll be pushing an update in a day or two that will allow for easy automatic conversion to a more performant file type.

**Comment by aneeshbal** [comment]
I see, that clears things up for point 1. As for point 2, I do not notice anything actually maxing out, but I am using mp4 format. I can try a different format, should not be too difficult to convert.

Thanks for the response and the help!

**Comment by jbohnslav** [comment]
Sure thing. The reason that .mp4 format is potentially the bottleneck is that image decompression and augmentation is done on the CPU. Depending on the codec on the .mp4, you might actually have to load multiple frames before the frame you're trying to load. This makes it very expensive to randomly grab frames from throughout a movie. 

If you convert it to a directory of .png files instead, with `deepethogram.file_io.VideoReader` and `VideoWriter`, random seeking is dramatically cheaper, at the expense of taking up more disk space.

**Comment by eliethesaiyan** [comment]
@jbohnslav , is the default number of epochs the same for flow Generator,  Feature extractor and sequence model?
